src/OgAccess.php

```diff
@@ -140,22 +166,30 @@ public function userAccess(EntityInterface $group, $operation, AccountInterface
       $permissions = [];
       $user_is_group_admin = FALSE;
 
-      if ($membership = Og::getMembership($group, $user)) {
-        foreach ($membership->getRoles() as $role) {
-          // Check for the is_admin flag.
-          /** @var \Drupal\og\Entity\OgRole $role */
-          if ($role->isAdmin()) {
-            $user_is_group_admin = TRUE;
-            break;
+      $states = [
+        OgMembershipInterface::STATE_ACTIVE,
+        OgMembershipInterface::STATE_PENDING,
+        OgMembershipInterface::STATE_BLOCKED,
+      ];
+      if ($membership = Og::getMembership($group, $user, $states)) {
+        // Blocked users don't have any permissions.
+        if ($membership->getState() !== OgMembershipInterface::STATE_BLOCKED) {
+          foreach ($membership->getRoles() as $role) {
+            // Check for the is_admin flag.
+            /** @var \Drupal\og\Entity\OgRole $role */
+            if ($role->isAdmin()) {
+              $user_is_group_admin = TRUE;
+              break;
+            }
+
+            $permissions = array_merge($permissions, $role->getPermissions());
           }
-
-          $permissions = array_merge($permissions, $role->getPermissions());
         }
       }
       else {
         // User is a non-member.
         /** @var \Drupal\og\Entity\OgRole $role */
-        $role = Og::getRole($group_type_id, $bundle, OgRoleInterface::ANONYMOUS);
+        $role = OgRole::loadByGroupAndName($group, OgRoleInterface::ANONYMOUS);
```

Comment:
👍 the name is fine for now - things might shift a bit once we'll have the perBundle/ perGroup roles.